### ISSUE TITLE
[Bug] [Integration] [argoCD] : failing to synchronize all apps

### ISSUE THREAD

**Issue by sgandon** [port-labs/ocean#577]
### Ocean Framework Version

🌊 Ocean version: 0.5.12

### Integration Version

🚢 Integration version: 0.1.38

### Steps to reproduce

launched the "docker run command"
may entities where ingested properly but one is failing

### What did you expect to see?

everything worked

### What did you see instead?

the following error
```
2024-04-21 09:30:22.551 | INFO     | Sending request to ArgoCD API: GET https://readacted.domain.com/api/v1/applications/messaging/managed-resources
2024-04-21 09:30:22.594 | ERROR    | Failed to execute resync function, error: 'items'
Traceback (most recent call last):
  File "/usr/local/bin/ocean", line 8, in <module>
    sys.exit(cli_start())
  File "/usr/local/lib/python3.11/site-packages/click/core.py", line 1157, in __call__
    return self.main(*args, **kwargs)
  File "/usr/local/lib/python3.11/site-packages/click/core.py", line 1078, in main
    rv = self.invoke(ctx)
  File "/usr/local/lib/python3.11/site-packages/click/core.py", line 1688, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/usr/local/lib/python3.11/site-packages/click/core.py", line 1434, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/usr/local/lib/python3.11/site-packages/click/core.py", line 783, in invoke
    return __callback(*args, **kwargs)
  File "/usr/local/lib/python3.11/site-packages/port_ocean/cli/commands/sail.py", line 74, in sail
    run(path, log_level, port, initialize_port_resources, override)
  File "/usr/local/lib/python3.11/site-packages/port_ocean/run.py", line 57, in run
    uvicorn.run(app, host="0.0.0.0", port=application_settings.port)
  File "/usr/local/lib/python3.11/site-packages/uvicorn/main.py", line 578, in run
    server.run()
  File "/usr/local/lib/python3.11/site-packages/uvicorn/server.py", line 61, in run
    return asyncio.run(self.serve(sockets=sockets))
  File "/usr/local/lib/python3.11/asyncio/runners.py", line 190, in run
    return runner.run(main)
  File "/usr/local/lib/python3.11/asyncio/runners.py", line 118, in run
    return self._loop.run_until_complete(task)
  File "/usr/local/lib/python3.11/asyncio/base_events.py", line 640, in run_until_complete
    self.run_forever()
  File "/usr/local/lib/python3.11/asyncio/base_events.py", line 607, in run_forever
    self._run_once()
  File "/usr/local/lib/python3.11/asyncio/base_events.py", line 1922, in _run_once
    handle._run()
  File "/usr/local/lib/python3.11/asyncio/events.py", line 80, in _run
    self._context.run(self._callback, *self._args)
  File "/usr/local/lib/python3.11/site-packages/port_ocean/core/integrations/mixins/sync_raw.py", line 196, in _register_in_batches
    async for items in generator:
  File "/usr/local/lib/python3.11/site-packages/port_ocean/core/integrations/mixins/utils.py", line 52, in resync_generator_wrapper
    with resync_error_handling():
  File "/usr/local/lib/python3.11/contextlib.py", line 155, in __exit__
    self.gen.throw(typ, value, traceback)
> File "/usr/local/lib/python3.11/site-packages/port_ocean/core/integrations/mixins/utils.py", line 23, in resync_error_handling
    yield
  File "/usr/local/lib/python3.11/site-packages/port_ocean/core/integrations/mixins/utils.py", line 53, in resync_generator_wrapper
    result = await anext(generator)
  File "/app/./main.py", line 47, in on_managed_resources_resync
    managed_resources = await argocd_client.get_managed_resources(
  File "/app/client.py", line 107, in get_managed_resources
    managed_resources = (await self._send_api_request(url=url))["items"]
Exception: 'items'
2024-04-21 09:30:22.600 | INFO     | Finished registering change for 1 raw results for kind: managed-resource. 736 entities were affected
2024-04-21 09:30:22.600 | INFO     | Starting resync diff calculation
2024-04-21 09:30:22.601 | ERROR    | Resync failed with 1. Skipping delete phase due to incomplete state
2024-04-21 09:30:22.601 | INFO     | Event finished
2024-04-21 09:30:22.602 | INFO     | Once event listener finished
2024-04-21 09:30:22.602 | INFO     | Exiting application
```

### Are you willing to submit a PR?

- [ ] I'm willing to submit a PR!

**Comment by PeyGis** [comment]
Thanks @sgandon Please note that a [PR](https://github.com/port-labs/ocean/pull/588) has been created for this issue

**Comment by yairsimantov20** [comment]
Should be resolved at version 0.1.40 for the argocd integration. #588
